### PR TITLE
chore: bump duckdb-rs version

### LINKED ISSUE #153: Geoparquet support
### What feature are you requesting?

Hi, I'm testing pg_analytics for mostly geospatial use cases. Does pg_analytics supports geoparquet format as Crunchy bridge does ?
I saw only GeoJSON examples in the docs and failed to do geographic subset filters when trying to mimic this example : https://www.crunchydata.com/blog/vehicle-routing-with-postgis-and-overture-data

### Why are you requesting this feature?

For a massive datawarehouse postgres with mostly geospatial data, we go toward analytics and explore all the available options to outsource storage of data to S3 geoparquet, and benefit from column storage features for analytics.  
After having screened most available extensions, pg_analytics seems really advanced and easy to set up.  
GeoParquet is now an Open Geospatial Consortium standard (OGC) and along with GeoPackage should be the first class citizens when dealing with spatial data.  

### What is your proposed implementation for this feature?

GDAL and duckDB both support GeoParquet reading and could provide the necessary bits. 

### Full Name:

Régis Haubourg

### Affiliation:

CSTB

### PR DESCRIPTION
**Pull request by DeoLeung**
# Ticket(s) Closed

- Closes #153 - 

## What

support reading and writing geoparquet

## Why

with `duckdb_spatial>=v1.1.0` geometry column in parquet will be read/write as geometry instead of blob

## How

no explicit change of sql is needed

## Tests

a read/write parquet test should be added later

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -24,7 +24,7 @@ chrono = "0.4.38"
 duckdb = { git = "https://github.com/paradedb/duckdb-rs.git", features = [
   "bundled",
   "extensions-full",
-], rev = "e532dd6" }
+], rev = "c2f9e2010e326de21126e90dc24da47e0a962cb0" }
```
Comment (philippemnoel) [inline Cargo.toml:27]: Where is this commit from?

**Comment by philippemnoel** [comment]
This PR would need documentation so people know how to use it

File: Cargo.toml (hunk 1 above)
Comment (philippemnoel) [reply Cargo.toml:27]: Note that we use our own fork of duckdb-rs. It is up-to-date, though.

Comment (DeoLeung) [reply Cargo.toml:27]: yes, it's the latest commit from paradedb/duckdb-rs

**Comment by DeoLeung** [comment]
> This PR would need documentation so people know how to use it

just fix the packages and now i can build it and test it locally

it seems there's no `geometry` type cell.rs, currently it still read the geometry column as text :(

**Comment by philippemnoel** [comment]
> > This PR would need documentation so people know how to use it
> 
> just fix the packages and now i can build it and test it locally
> 
> it seems there's no `geometry` type cell.rs, currently it still read the geometry column as text :(

Yes, that’s because duckdb-rs is on duckdb 1.0.0, not 1.1.0 which it seems like it needs. We would need you to submit a PR to our fork to duckdb-rs to upgrade it as well there, then update the commit here.

**Comment by DeoLeung** [comment]
> > > This PR would need documentation so people know how to use it
> > 
> > 
> > just fix the packages and now i can build it and test it locally
> > it seems there's no `geometry` type cell.rs, currently it still read the geometry column as text :(
> 
> Yes, that’s because duckdb-rs is on duckdb 1.0.0, not 1.1.0 which it seems like it needs. We would need you to submit a PR to our fork to duckdb-rs to upgrade it as well there, then update the commit here.

from https://github.com/paradedb/duckdb-rs/tree/main/crates/libduckdb-sys it seems it's already on `v1.1.1`

**Comment by philippemnoel** [comment]
> > > > This PR would need documentation so people know how to use it
> > > 
> > > 
> > > just fix the packages and now i can build it and test it locally
> > > it seems there's no `geometry` type cell.rs, currently it still read the geometry column as text :(
> > 
> > 
> > Yes, that’s because duckdb-rs is on duckdb 1.0.0, not 1.1.0 which it seems like it needs. We would need you to submit a PR to our fork to duckdb-rs to upgrade it as well there, then update the commit here.
> 
> from https://github.com/paradedb/duckdb-rs/tree/main/crates/libduckdb-sys it seems it's already on `v1.1.1`

Ah yes! You are right! Oops, I had no idea. Then, I'm not sure... Are you sure this is supported? Do we need to do anything to enable it?

**Comment by DeoLeung** [comment]
> Could you please sign the CLA and I think this will be good to go :)

done :)

for the geoparquet reference, i think it need some extra data type mapping (especially postgis geometry)

https://github.com/duckdb/duckdb/pull/12503
https://github.com/duckdb/pg_duckdb/issues/141

**Comment by philippemnoel** [comment]
> > Could you please sign the CLA and I think this will be good to go :)
> 
> done :)
> 
> for the geoparquet reference, i think it need some extra data type mapping (especially postgis geometry)
> 
> [duckdb/duckdb#12503](https://github.com/duckdb/duckdb/pull/12503) [duckdb/pg_duckdb#141](https://github.com/duckdb/pg_duckdb/issues/141)

We ship with Postgis support and the spatial extension is enabled in `pg_analytics`. It was added a while back. Does that give you enough?

**Comment by DeoLeung** [comment]
> > > Could you please sign the CLA and I think this will be good to go :)
> > 
> > 
> > done :)
> > for the geoparquet reference, i think it need some extra data type mapping (especially postgis geometry)
> > [duckdb/duckdb#12503](https://github.com/duckdb/duckdb/pull/12503) [duckdb/pg_duckdb#141](https://github.com/duckdb/pg_duckdb/issues/141)
> 
> We ship with Postgis support and the spatial extension is enabled in `pg_analytics`. It was added a while back. Does that give you enough?

not enough at the moment. 
we want to store the analytic data into s3 in parquet format, most of them are gis data, with newly duckdb, we could dump them as geoparquet, and read it back directly as geometry data type
```sql
load spatial;
create table gis_test as select st_makepoint(127.0, 30.0) as geometry;
copy  gis_test to '~/Downloads/gis_test.parquet' (format parquet);
create table gis2 as select * from '~/Downloads/gis_test.parquet';
describe table gis2;
┌─────────────┬─────────────┬─────────┬─────────┬─────────┬─────────┐
│ column_name │ column_type │  null   │   key   │ default │  extra  │
│   varchar   │   varchar   │ varchar │ varchar │ varchar │ varchar │
├─────────────┼─────────────┼─────────┼─────────┼─────────┼─────────┤
│ geometry    │ GEOMETRY    │ YES     │         │         │         │
└─────────────┴─────────────┴─────────┴─────────┴─────────┴─────────┘
```

but in pg_analytics, it still read it as text column
```sql
CREATE FOREIGN TABLE gis2 () server parquet_server options(files '~/Downloads/gis_test.parquet');
select * from information_schema.columns where table_schema = 'public' and table_name = 'gis2' and column_name = 'geometry';
select column_name, data_type from information_schema.columns where table_schema = 'public' and table_name = 'gis2' and column_name = 'geometry';
 column_name | data_type 
-------------+-----------
 geometry    | text
(1 row)
```

**Comment by philippemnoel** [comment]
> > > > Could you please sign the CLA and I think this will be good to go :)
> > > 
> > > 
> > > done :)
> > > for the geoparquet reference, i think it need some extra data type mapping (especially postgis geometry)
> > > [duckdb/duckdb#12503](https://github.com/duckdb/duckdb/pull/12503) [duckdb/pg_duckdb#141](https://github.com/duckdb/pg_duckdb/issues/141)
> > 
> > 
> > We ship with Postgis support and the spatial extension is enabled in `pg_analytics`. It was added a while back. Does that give you enough?
> 
> not enough at the moment. we want to store the analytic data into s3 in parquet format, most of them are gis data, with newly duckdb, we could dump them as geoparquet, and read it back directly as geometry data type
> 
> ```sql
> load spatial;
> create table gis_test as select st_makepoint(127.0, 30.0) as geometry;
> copy  gis_test to '~/Downloads/gis_test.parquet' (format parquet);
> create table gis2 as select * from '~/Downloads/gis_test.parquet';
> describe table gis2;
> ┌─────────────┬─────────────┬─────────┬─────────┬─────────┬─────────┐
> │ column_name │ column_type │  null   │   key   │ default │  extra  │
> │   varchar   │   varchar   │ varchar │ varchar │ varchar │ varchar │
> ├─────────────┼─────────────┼─────────┼─────────┼─────────┼─────────┤
> │ geometry    │ GEOMETRY    │ YES     │         │         │         │
> └─────────────┴─────────────┴─────────┴─────────┴─────────┴─────────┘
> ```
> 
> but in pg_analytics, it still read it as text column
> 
> ```sql
> CREATE FOREIGN TABLE gis2 () server parquet_server options(files '~/Downloads/gis_test.parquet');
> select * from information_schema.columns where table_schema = 'public' and table_name = 'gis2' and column_name = 'geometry';
> select column_name, data_type from information_schema.columns where table_schema = 'public' and table_name = 'gis2' and column_name = 'geometry';
>  column_name | data_type 
> -------------+-----------
>  geometry    | text
> (1 row)
> ```

Can you confirm the duckDB version you are using separately? And can you confirm the GIS extension is loaded in pg_analytics?

**Comment by DeoLeung** [comment]
confirm duckdb version `1.1.1` and `1.1.3` works with geoparquet

confirm `postgis` is loaded in postgres
but I'm not sure how to check if duckdb `spatial` is enabled  in pg_analytics :(

tried `select duckdb_execute($$load spatial$$);` with no luck

**Review by philippemnoel — APPROVED** [review]

**Pull request merged**